### PR TITLE
RecoTracker/TransientTrackingRecHit: change return type of ESProducers.

### PR DESCRIPTION
**Pull request by gartung**
Remove shared_ptr class member not needed.

### CODE REVIEW

**Comment by gartung** [comment]
@Dr15Jones

File: RecoTracker/TransientTrackingRecHit/plugins/TkTransientTrackingRecHitBuilderESProducer.cc [hunk 1]
```diff
@@ -89,12 +89,11 @@ TkTransientTrackingRecHitBuilderESProducer::produce(const TransientRecHitRecord
   if (p2OTname != "") {
     iRecord.getRecord<TkStripCPERecord>().get( p2OTname, p2OTe );
     p2OTp = p2OTe.product();
-    _builder  = std::make_shared<TkTransientTrackingRecHitBuilder>(pDD.product(), pp, p2OTp);
+    return std::make_unique<TkTransientTrackingRecHitBuilder>(pDD.product(), pp, p2OTp);
   } else {
-    _builder  = std::make_shared<TkTransientTrackingRecHitBuilder>(pDD.product(), pp, sp, mp, computeCoarseLocalPositionFromDisk);
+    return std::make_unique<TkTransientTrackingRecHitBuilder>(pDD.product(), pp, sp, mp, computeCoarseLocalPositionFromDisk);
```
Comment (Dr15Jones) [inline RecoTracker/TransientTrackingRecHit/plugins/TkTransientTrackingRecHitBuilderESProducer.cc]: You can remove the else and just return what is in that block

**Review by Dr15Jones — CHANGES_REQUESTED** [review]

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by perrotta** [comment]
+1
- technical
- jenkins tests pass

**Comment by fabiocos** [comment]
+1

**Pull request merged**